sopel/modules/weather.py

```diff
@@ -17,52 +17,50 @@ def woeid_search(query):
     node for the result, so that location data can still be retrieved. Returns
     None if there is no result, or the woeid field is empty.
     """
-    query = 'q=select * from geo.places where text="%s"' % query
-    body = requests.get('https://query.yahooapis.com/v1/public/yql?' + query)
-    parsed = xmltodict.parse(body.text).get('query')
-    results = parsed.get('results')
-    if results is None or results.get('place') is None:
+    query = query.replace(" ", "+")
+    body = requests.get('https://www.metaweather.com/api/location/search/?query=' + query)
+    results = body.json()
```

Comment:
This might fail if the status is not 200 or if the content is not JSON. You may want to add at least a try/except block around this.